public/app/core/components/NavBar/Next/NavBarScrollContainer.tsx

```diff
@@ -0,0 +1,126 @@
+  const scrollBottomMarker = useRef<HTMLDivElement>(null);
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  // Here we observe the top and bottom markers to determine if we should show the scroll indicators
+  useEffect(() => {
+    const intersectionObserver = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.target === scrollTopMarker.current) {
+          setShowTopScrollIndicator(!entry.isIntersecting);
+        } else if (entry.target === scrollBottomMarker.current) {
+          setShowBottomScrollIndicator(!entry.isIntersecting);
+        }
+      });
+    });
+    [scrollTopMarker, scrollBottomMarker].forEach((ref) => {
+      if (ref.current) {
+        intersectionObserver.observe(ref.current);
+      }
+    });
+    return () => intersectionObserver.disconnect();
+  }, []);
+
+  return (
+    <CustomScrollbar className={styles.scrollContainer} hideVerticalTrack hideHorizontalTrack>
+      <div
+        className={cx(styles.scrollTopIndicator, {
+          [styles.scrollIndicatorVisible]: showScrollTopIndicator,
+        })}
+      >
+        <Icon className={styles.scrollTopIcon} name="angle-up" />
+      </div>
+      <div className={styles.scrollContent}>
+        <div className={styles.scrollTopMarker} ref={scrollTopMarker}></div>
+        {children}
+        <div className={styles.scrollBottomMarker} ref={scrollBottomMarker}></div>
+      </div>
+      <div
+        className={cx(styles.scrollBottomIndicator, {
+          [styles.scrollIndicatorVisible]: showScrollBottomIndicator,
+        })}
+      >
+        <Icon className={styles.scrollBottomIcon} name="angle-down" />
+      </div>
+    </CustomScrollbar>
+  );
+};
+
+NavBarScrollContainer.displayName = 'NavBarScrollContainer';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  'scrollTopMarker, scrollBottomMarker': css({
+    height: theme.spacing(1),
+    left: 0,
+    position: 'absolute',
+    pointerEvents: 'none',
+    right: 0,
+  }),
+  scrollTopMarker: css({
+    top: 0,
+  }),
+  scrollBottomMarker: css({
+    bottom: 0,
+  }),
+  scrollContent: css({
+    position: 'relative',
+  }),
+  scrollContainer: css`
+    .scrollbar-view {
+      position: static !important;
+    }
+  `,
+  scrollTopIndicator: css({
+    background: `linear-gradient(0deg, transparent, ${theme.colors.background.canvas})`,
+    height: theme.spacing(6),
+    left: 0,
+    opacity: 0,
+    pointerEvents: 'none',
+    position: 'absolute',
+    right: 0,
+    top: 0,
+    transition: theme.transitions.create('opacity'),
+    zIndex: theme.zIndex.sidemenu,
+  }),
+  scrollBottomIndicator: css({
+    background: `linear-gradient(0deg, ${theme.colors.background.canvas}, transparent)`,
+    bottom: 0,
+    height: theme.spacing(6),
+    left: 0,
+    opacity: 0,
+    pointerEvents: 'none',
+    position: 'absolute',
+    right: 0,
+    transition: theme.transitions.create('opacity'),
+    zIndex: theme.zIndex.sidemenu,
+  }),
```

Comment:
Could you do something similar to what you did for `'scrollTopMarker, scrollBottomMarker'` and join the common styling and just keep what is different separate? I find it easier to understand then what are the differences between the two

Reply:
@JoaoSilvaGrafana i tried that but it doesn't work - i actually don't know why tbh as it was exactly the same syntax as the other... 😢

Reply:
Weird! Nevermind, it was just a small thing anyway